src/Twitter.php

```diff
@@ -137,17 +145,19 @@ public function usingConfiguration(Configuration $configuration): self
      * @return mixed|string
      */
     public function query(
-        string $name,
+        string $endpoint,
         string $requestMethod = self::REQUEST_METHOD_GET,
         array $parameters = [],
         bool $multipart = false,
         string $extension = self::DEFAULT_EXTENSION
     ) {
         try {
-            $this->logRequest($name, $requestMethod, $parameters, $multipart);
+            $this->logRequest($endpoint, $requestMethod, $parameters, $multipart);
 
             $host = !$multipart ? $this->config->getApiUrl() : $this->config->getUploadUrl();
-            $url = $this->buildUrl($host, $this->config->getApiVersion(), $name, $extension);
+            $url = isset($parameters[self::KEY_PREBUILT_ENDPOINT]) && $parameters[self::KEY_PREBUILT_ENDPOINT]
```

Comment:
I think it would make sense to split this out into a seperate method for direct queries.

If we wanted to reduce duplication, we could have a function to make the actual request (ie, taking the complete URL). This function could then be switched to just build the URL, and then pass it to the request function.

This might also remove the need for setting the credentials to empty at the start?

Reply:
Agreed. Will look into this.